### PR TITLE
262 password setup form

### LINKED ISSUE #262: [FRONTEND] Create password setup form
**Is your feature request related to a problem? Please describe.**

Blocked by #228 
we assume the user is logged in so auto fill their email (hint: use the `useAppData` hook from sweet state)

CREATE COMPONENT, ADD LOGIC, ADD ROUTE

Image

[We should use this function:](https://firebase.google.com/docs/auth/web/manage-users#set_a_users_password)
```ts 
import { updatePassword } from "firebase/auth";

updatePassword(//user from sweet state, newPassword).then(() => {
  // Update successful.
}).catch((error) => {
  // An error ocurred
  // ...
});
```
**BEFORE MERGING**

- [ ] Storybooks created where possible
- [ ] Tested with [nock](https://tanstack.com/query/v3/docs/framework/react/guides/testing) if data fetching
- [ ] Tests written for critical interactions
- [ ] PR Reviewed (For non-trivial changes)
- [ ] Changes tested after rebasing on master or merging in master (_hint_: `git fetch origin master:master`, then `git rebase master` or `git merge master`)
- [ ] All required PR checks passing

### PR DESCRIPTION
**Pull request by zlrkw11**
closes #262

### CODE REVIEW

**Comment by zlrkw11** [comment]
image

File: client/src/components/generic/PasswordSetup/PasswordSetup.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+              }
+              type="password"
+              pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
+              title="Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters"
+            />
+          </div>
+          <div className="">
+            <p>Confirm password</p>
+            <TextInput
+              className="w-3/4 rounded"
```
Comment (bcho892) [inline client/src/components/generic/PasswordSetup/PasswordSetup.tsx:97]: can we make it full width on smaller screens?

File: client/src/components/generic/PasswordSetup/PasswordSetup.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+              type="password"
+              onChange={(e) =>
+                setFormData({ ...formData, secondPassword: e.target.value })
+              }
+            />
+          </div>
+        </div>
+        <p className="pt-[48px]">You can reset your password at any time.</p>
```
Comment (bcho892) [inline client/src/components/generic/PasswordSetup/PasswordSetup.tsx:105]: feels like too much padding

**Review by bcho892 — COMMENTED** [review]
looks mostly good on styling side, ill add some tests tonight and we can get this merged in

File: client/src/components/generic/PasswordSetup/PasswordSetup.tsx (hunk 2 above)
Comment (zlrkw11) [reply client/src/components/generic/PasswordSetup/PasswordSetup.tsx]: changed to 24px to balance out after calculation

**Comment by zlrkw11** [comment]
image
- full input boxes as requested on small screen size

**Comment by zlrkw11** [comment]
image
- default desktop screen with adjusted spacing

File: client/src/components/generic/PasswordSetupForm/PasswordSetupForm.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        })
+      }
+    } catch (e) {
+      console.error(e)
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <div className="">
```
Comment (bcho892) [inline client/src/components/generic/PasswordSetupForm/PasswordSetupForm.tsx:78]: nit: avoid empty class names

**Review by bcho892 — APPROVED** [review]

**Pull request merged**